python_modules/dagster/dagster/_core/scheduler/instigation.py

```diff
@@ -452,12 +460,23 @@ def requested_assets_and_partitions(self) -> Mapping[AssetKey, AbstractSet[str]]
 
         partitions_by_asset_key = {}
         for run_request in self.tick_data.run_requests or []:
-            for asset_key in run_request.asset_selection or []:
-                if asset_key not in partitions_by_asset_key:
-                    partitions_by_asset_key[asset_key] = set()
-
-                if run_request.partition_key:
-                    partitions_by_asset_key[asset_key].add(run_request.partition_key)
+            if run_request.requires_backfill_daemon():
+                for (
+                    asset_key_partition_key
+                ) in run_request.asset_graph_subset.iterate_asset_partitions():
```

Comment:
i think it would basically be the same - both require a call to `partitions_subset.get_partition_keys` so it seems like the only difference would be perf of `add` vs `update`